app/samples/src/main/java/com/sygic/samples/search/SearchFromBrowseMapWithPinsActivity.kt

```diff
@@ -0,0 +1,112 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.samples.search
+
+import android.os.Bundle
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.sygic.maps.module.browsemap.BROWSE_MAP_FRAGMENT_TAG
+import com.sygic.maps.module.browsemap.BrowseMapFragment
+import com.sygic.maps.module.common.provider.ModuleConnectionProvider
+import com.sygic.samples.R
+import com.sygic.samples.app.activities.CommonSampleActivity
+import com.sygic.samples.search.viewmodels.SearchFromBrowseMapWitchPinsActivityViewModel
+import com.sygic.sdk.map.MapRectangle
+import com.sygic.sdk.position.GeoCoordinates
+import com.sygic.sdk.map.`object`.MapMarker
+
+class SearchFromBrowseMapWithPinsActivity : CommonSampleActivity() {
+
+    override val wikiModulePath: String = "Module-Search#search---from-browse-map-with-pins"
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_search_from_browse_map_pins)
+
+        val viewModel =
+            ViewModelProviders.of(this).get(SearchFromBrowseMapWitchPinsActivityViewModel::class.java).apply {
+                this.addMapMarkerObservable.observe(
+                    this@SearchFromBrowseMapWithPinsActivity,
+                    Observer<MapMarker> { addMapMarker(it) })
+                this.removeAllMapMarkersObservable.observe(
+                    this@SearchFromBrowseMapWithPinsActivity,
+                    Observer<Any> { removeAllMapMarkers() })
+                this.setCameraPositionObservable.observe(
+                    this@SearchFromBrowseMapWithPinsActivity,
+                    Observer<GeoCoordinates> { setCameraPosition(it) })
+                this.setCameraRectangleObservable.observe(
+                    this@SearchFromBrowseMapWithPinsActivity,
+                    Observer<MapRectangle> { setCameraRectangle(it) })
+                this.setCameraZoomLevelObservable.observe(
+                    this@SearchFromBrowseMapWithPinsActivity,
+                    Observer<Float> { setCameraZoomLevel(it) })
+            }
+
+        if (savedInstanceState == null) {
+            setFragmentModuleConnection(placeBrowseMapFragment(2F), viewModel)
+        } else {
+            setFragmentModuleConnection(
+                supportFragmentManager.findFragmentByTag(BROWSE_MAP_FRAGMENT_TAG) as BrowseMapFragment, viewModel
+            )
+        }
+    }
+
+    // Note: You can also create this Fragment just like in other examples directly in an XML layout file, but
+    // performance or other issues may occur (https://stackoverflow.com/a/14810676/3796931).
+    private fun placeBrowseMapFragment(zoomLevel: Float) =
```

Comment:
ako je to hned vedla sena, ale keby som nepozrel co je 2F, tak by som neuhadol ze zoom.

Skor by som ten apply block dal hore do `onCreate` za volanie tejto metody